src/lib/sql/sql.cpp

```diff
@@ -0,0 +1,43 @@
+#include "sql.hpp"
+
+namespace opossum {
+
+SQL::SQL(const std::string& sql) : _sql(sql) {}
+
+SQL& SQL::set_use_mvcc(const UseMvcc use_mvcc) {
+  _use_mvcc = use_mvcc;
+  return *this;
+}
+
+SQL& SQL::set_optimizer(const std::shared_ptr<Optimizer>& optimizer) {
+  _optimizer = optimizer;
+  return *this;
+}
+
+SQL& SQL::set_prepared_statement_cache(const PreparedStatementCache& prepared_statements) {
+  _prepared_statements = prepared_statements;
+  return *this;
+}
+
+SQL& SQL::set_transaction_context(const std::shared_ptr<TransactionContext>& transaction_context) {
+  _transaction_context = transaction_context;
+  _use_mvcc = UseMvcc::Yes;
+
+  return *this;
+}
+
+SQL& SQL::disable_mvcc() { return set_use_mvcc(UseMvcc::No); }
```

Comment:
Why do we have this and `set_use_mccc`?

Reply:
disable_mvcc() is done a lot, so it seemed convenient. Will remove it if you want to. Will doc it if you want to.